packages/dapp/src/hooks/useKeysManager.ts

```diff
@@ -0,0 +1,156 @@
+import type { GenericStateRecord } from '../store';
+import { useState, useCallback } from 'react';
+import { useAppReducer } from '../store/reducer';
+
+export const allowedKeysTypes: string[] = ['EcdsaSecp256k1VerificationKey2019'];
+
+export type KeyType =
+	typeof allowedKeysTypes[number];
+
+export enum RevocationReason {
+  "unspecified",
+  "keyCompromise",
+  "affiliationChanged",
+  "superseded",
+  "cessationOfOperation",
+  "certificateHold",
+  "privilegeWithdrawn"
+}
+
+export interface KeyRecord extends GenericStateRecord {
+	id: string; // Unique Id generated using `simpleUid()` 
+							// from @windingtree/org.id-utils package. 
+							// For example see the docs: 
+							// https://windingtree.github.io/org.id-sdk/#/build/utils/index?id=simpleuidlength
+	type: KeyType;
+	publicKey: string; // Ethereum account address
+	tag: string; // Unique across records formatted string
+	note?: string; // Key description
+	revocation?: VerificationMethodReference['verificationMethodRevocation']; // see this type for details
+}
+
+export type Keys = KeyRecord[];
+
+export type UseKeysManager = [
+  addKey: Function,
+  updateKey: Function,
+  removeKey: Function,
+  revokeKey: Function,
+  loading: boolean,
+  error: string | undefined,
+];
+
+// UseKeysManager react hook
+export const useKeysManager = (): UseKeysManager => {
+  const [state, dispatch] = useAppReducer();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const findKeyById = (id:string) => (state.keys as Keys).find((key) => key.id == id)
+  const findKeyByTag = (tag:string) => (state.keys as Keys).find((key) => key.tag == tag)
+
+  const addKey = useCallback((record:KeyRecord) => {
+    setLoading(true);
+    let keyExist = findKeyById(record.id)
+    if (keyExist !== undefined) {
+      setError('Provided key already exists')
+      setLoading(false);
+      return 
+    }
+    let tagExist = findKeyByTag(record.tag)
+    if (tagExist !== undefined) {
+      setError('Provided tag already used, pick another one')
+      setLoading(false);
+      return 
+    }
+
+    dispatch({
+      type: 'SET_RECORD',
+      payload: {
+        name: 'keys',
+        record: record
+      }
+    });
+    setLoading(false);
+  }, []);
+
+  const updateKey = useCallback((record:KeyRecord) => {
+    setLoading(true);
+    let keyExist = findKeyById(record.id)
+    if (keyExist === undefined) {
+      setError('Provided key does not have corresponding id')
+      setLoading(false);
+      return 
+    }
+    if(keyExist.publicKey !== record.publicKey) {
+      setError('PublicKey is not allowed to be changed')
+      setLoading(false);
+      return
+    }
+    if(keyExist.type !== record.type) {
+      setError('Type is not allowed to be changed')
+      setLoading(false);
+      return
+    }
+    if(record.revocation !== undefined && keyExist.revocation !== record.revocation) {
```

Comment:
the same as https://github.com/windingtree/org.id-profile-manager/pull/7/files#r805971075

Reply:
✔️